File: Cpp/1828_Queries_on_Number_of_Points_Inside_a_Circle.cpp
class Solution {
public:
    vector<int> countPoints(vector<vector<int>>& points, vector<vector<int>>& queries) {
        vector<int> result;
        for(int i = 0; i < queries.size(); i++){
            int dr = queries[i][2] * queries[i][2];
            int temp = 0;
            for(int j = 0; j < points.size(); j++){
                int dx = queries[i][0] - points[j][0];
                int dy = queries[i][1] - points[j][1];
                int dxy = (dx * dx) + (dy * dy);
                    if(dxy <= dr){
                    temp++;
                }
            }
            result.push_back(temp);
        }
        /*
        for(auto query : queries){
            int dr = query[2] * query[2];
            int temp = 0;
            for(auto point : points){
                int dxy = (query[0] - point[0]) * (query[0] - point[0]) 
                            + (query[1] - point[1]) * (query[1] - point[1]);
                if(dxy <= dr){
                    temp++;
                }
            }
            result.push_back(temp);
        }
        */
        return result;
    }
};

File: Python3/1828_Queries_on_Number_of_Points_Inside_a_Circle.py
class Solution:
    def countPoints(self, points: List[List[int]], queries: List[List[int]]) -> List[int]:
        def isInside(x1, y1, x, y, r):
            return (x1-x) * (x1-x) + (y1-y)*(y1-y) <= r*r
        
        result = []
        for x, y, r in queries:
            t = 0
            for x1, y1 in points:
                if isInside(x1, y1, x, y, r):
                    t += 1
            result.append(t)
        
        return result
        
        """
        result = [0] * len(queries)
        
        for i in range(0, len(queries)):
            for p in points:
                if (p[0] - queries[i][0]) * (p[0] - queries[i][0]) + (p[1] - queries[i][1]) * (p[1] - queries[i][1]) <= (queries[i][2]) * (queries[i][2]) :
                    result[i] = result[i] + 1
                    
        return result
        """
                    
